storage/types.go

```diff
@@ -156,16 +168,19 @@ func (n *NodeID) SetBit(i int, b uint) {
 
 // Bit returns 1 if the ith bit is true, and false otherwise.
 func (n *NodeID) Bit(i int) uint {
-	bIndex := (n.PathLenBits - i - 1) / 8
+	if got, want := i, n.PathLenBits()-1; got > want {
+		panic(fmt.Sprintf("storage: Bit(%v) > (PathLenBits -1): %v", got, want))
```

Comment:
I found the panics slightly easier to read, and a nice documentation of the exact preconditions the function expects. 

There are a number of cases where the bounds check would succeed and produce garbage data, whereas this check would catch the logic bug. For example, imagine a case where `PathLenBits` is not a multiple of 8, say 7.  One could request `Bit(8)` without causing an out-of-bounds error, but the result may still be undefined.  

What do you advise?